modules/core/04-channel/keeper/packet_test.go

```diff
@@ -940,3 +988,74 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
 		})
 	}
 }
+
+// TestAcknowledgeFlush tests that Acknowledging the last in-flight packet moves the channel
+// state to FLUSHINGCOMPLETE.
+func (suite *KeeperTestSuite) TestAcknowledgeFlushStatus() {
+	var path *ibctesting.Path
+
+	testCases := []struct {
+		msg                 string
+		malleate            func()
+		expectedFlushStatus types.FlushStatus
+	}{
+		{
+			"success: final packet commitment cleared, flush status set to FLUSHCOMPLETE",
+			func() {},
+			types.FLUSHCOMPLETE,
+		},
+		{
+			"success: has in-flight packets, flush status remains FLUSHING",
+			func() {
+				// Send an additional packet
+				sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+				suite.Require().Equal(uint64(1), sequence)
+				suite.Require().NoError(err)
+			},
+			types.FLUSHING,
+		},
+	}
+
+	for i, tc := range testCases {
+		tc := tc
+		suite.Run(fmt.Sprintf("Case %s, %d/%d tests", tc.msg, i, len(testCases)), func() {
+			suite.SetupTest() // reset
+			path = ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			tc.malleate()
+
+			// create packet commitment
+			sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+			suite.Require().NoError(err)
+
+			// create packet receipt and acknowledgement
+			packet := types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+			err = path.EndpointB.RecvPacket(packet)
+			suite.Require().NoError(err)
+
+			// Move channel to UPGRADE_ACK, flush status set to flushing due to previous SendPacket
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = ibcmock.UpgradeVersion
+
+			err = path.EndpointA.ChanUpgradeInit()
+			suite.Require().NoError(err)
+
+			err = path.EndpointB.ChanUpgradeTry()
+			suite.Require().NoError(err)
+
+			err = path.EndpointA.ChanUpgradeAck()
+			suite.Require().NoError(err)
+
+			packetKey := host.PacketAcknowledgementKey(packet.GetDestPort(), packet.GetDestChannel(), packet.GetSequence())
+			proof, proofHeight := path.EndpointB.QueryProof(packetKey)
+
+			channelCap := suite.chainA.GetChannelCapability(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+			err = suite.chainA.App.GetIBCKeeper().ChannelKeeper.AcknowledgePacket(suite.chainA.GetContext(), channelCap, packet, ibcmock.MockAcknowledgement.Acknowledgement(), proof, proofHeight)
+			suite.Require().NoError(err)
+
+			// Check that we've moved to FLUSHCOMPLETE
```

Comment:
This comment is not true for the second test case, right?

Reply:
yeah! should've been 'moved to expected flush status'. a relic from when this test only checked a single case.